src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -149,6 +156,9 @@ Score(const size_t queryIndex, TreeType& referenceNode)
     // Don't explore this tree branch.
     score = DBL_MAX;
 
+    // Update accumulated unused error tolerance.
+    accumError(queryIndex) = std::max(accumError(queryIndex) - bound, 0.0);
```

Comment:
Hmm, I think it is possible to do a little more pruning here.  What this code does (if I read it right) is subtracts all the error that was "used" by this node (as represented by `bound`) from `accumError(queryIndex)`.  But actually we have a little extra error as given by `errorTolerance`.

So, if we do this instead, does it work?

```
accumError(queryIndex) -= std::max(bound - errorTolerance, 0.0);
```

That instead subtracts any "extra" error used above and beyond the original error tolerance.

Reply:
In fact, to follow onto this, it may be possible to do

```
accumError(queryIndex) -= (bound - errorTolerance);
```

because if `bound - errorTolerance` is negative, this means that the bound we had was *within* the error tolerance, and we have some extra error tolerance that we can use.

However, I have not worked out whether or not this will actually work in practice.